ultraplot/figure.py

```diff
@@ -1131,11 +1126,83 @@ def _add_subplot(self, *args, **kwargs):
         kwargs.setdefault("label", f"subplot_{self._subplot_counter}")
         kwargs.setdefault("number", 1 + max(self._subplot_dict, default=0))
         kwargs.pop("refwidth", None)  # TODO: remove this
+
         ax = super().add_subplot(ss, _subplot_spec=ss, **kwargs)
+        self._WARN_SHARING_GEOAXIS = True
+        # Allow sharing for GeoAxes if rectilinear
+        if self._sharex or self._sharey:
+            if (
+                len(self.axes) > 1
+                and isinstance(ax, paxes.GeoAxes)
+                and not ax._is_rectilinear()
+            ):
+                self._unshare_axes()
+                if self._WARN_SHARING_GEOAXIS:
+                    warnings._warn_ultraplot(
+                        f"GeoAxes can only be shared for rectilinear projections, {ax.projection=} is not a rectilinear projection."
+                    )
+                    # Only warn once. Note, if axes are reshared
+                    # the warning is not reset. This is however,
+                    # very unlikely to happen as GeoAxes are not
+                    # typically shared and unshared.
+                    self._WARN_SHARING_GEOAXIS = False
+
         if ax.number:
             self._subplot_dict[ax.number] = ax
         return ax
 
+    def _unshare_axes(self):
+        for which in "x y z".split():
+            self._toggle_axis_sharing(which=which, share=False)
+        # Force setting extent
+        # This is necessary to ensure that the axes are properly aligned and we don't get weird scaling issues for geographic axes. This action is expensive for GeoAxes
+        for ax in self.axes:
+            if isinstance(ax, paxes.GeoAxes) and hasattr(ax, "set_global"):
+                ax.set_global()
+
+    def _toggle_axis_sharing(
+        self,
+        *,
+        which="y",
+        share=True,
+        panels=False,
+        children=False,
+        hidden=False,
+    ):
+        """
+        Share or unshare axes in the figure along a given direction.
+
+        Parameters:
+        - which: 'x', 'y' or 'view'.
+        - share: int indicating the levels (see above)
+        - panels: Whether to include panel axes.
```

Comment:
Not all of the parameters are documented.

Reply:
Thanks